File: 1353-find-resultant-array-after-removing-anagrams/1353-find-resultant-array-after-removing-anagrams.py
class Solution:
    def removeAnagrams(self, words: List[str]) -> List[str]:
        def anagram(s : str, r : str) -> bool:
            if len(s) != len(r):
                return False
            
            freq = [0 for _ in range(26)]
            for i in range(len(s)):
                freq[ord(s[i]) - ord('a')] += 1
                freq[ord(r[i]) - ord('a')] -= 1
            
            for i in range(26):
                if freq[i] != 0:
                    return False
            
            return True

        n = len(words)
        res = [words[0]]
        for i in range(1, n):
            if anagram(words[i], words[i - 1]):
                continue
            res.append(words[i])
        
        return res
        

File: 1353-find-resultant-array-after-removing-anagrams/1353-find-resultant-array-after-removing-anagrams.cpp
class Solution {
private:
    bool isAna(string s, string t) {
        if (s.length() != t.length())
            return false;

        vector<int> fre(26, false);

        for (int i = 0; i < s.length(); i++) {
            fre[s[i] - 'a']++;
            fre[t[i] - 'a']--;
        }
        for (int i = 0; i < fre.size(); i++) {
            if (fre[i] != 0)
                return false;
        }
        return true;
    }

public:
    vector<string> removeAnagrams(vector<string>& words) {
        int n = words.size();
        vector<string> ans;
        ans.push_back(words[0]);
        for (int i = 1; i < n; ++i) {
            if (isAna(words[i], words[i - 1]))
                continue;

            ans.push_back(words[i]);
        }
        /*
        while (!q.empty()) {
            auto top = q.front();
            q.pop();
            ans.push_back(top);
        }*/
        return ans;
    }
};